threepid_checker/__init__.py

```diff
@@ -29,18 +31,68 @@ def __init__(self, config: ThreepidCheckerConfig, api: ModuleApi):
         self._api = api
         self._config = config
 
+        self._http_client = api.http_client
+
+        self._api.register_password_auth_provider_callbacks(
+            is_3pid_allowed=self.check_if_allowed,
+        )
+
     @staticmethod
     def parse_config(config: Dict[str, Any]) -> ThreepidCheckerConfig:
-        # Parse the module's configuration here.
-        # If there is an issue with the configuration, raise a
-        # synapse.module_api.errors.ConfigError.
-        #
-        # Example:
-        #
-        #     some_option = config.get("some_option")
-        #     if some_option is None:
-        #          raise ConfigError("Missing option 'some_option'")
-        #      if not isinstance(some_option, str):
-        #          raise ConfigError("Config option 'some_option' must be a string")
-        #
-        return ThreepidCheckerConfig()
+        """Checks that required config options are in the correct format and parses them
+        into a ThreepidCheckerConfig.
+
+        Args:
+            config: The raw configuration dict.
+
+        Returns:
+            The parsed configuration.
+        """
+        if "url" not in config:
+            raise ConfigError('"url" is a required configuration parameter')
+
+        if not isinstance(config["url"], str) or not config["url"].startswith("http"):
```

Comment:
I think we usually check if it starts with `http:` or `https:`, but this might be OK?